lib/matplotlib/backend_bases.py

```diff
@@ -138,6 +145,35 @@ def get_registered_canvas_class(format):
     return backend_class
 
 
+class MainLoopBase(object):
```

Comment:
Whatever you say, mpl-3.0 fine with me too. :wink:  I see us using 2.x to further develop the work started under the MEPs (new tools etcetera) and thus encourage conversion, then with mpl-3.0 we can shed all the obsolete code.

The GUI backends use ``Gcf`` to control the individual figures and to check if we can end the mainloop (i.e. if we have all figures closed).  The Pdf backend uses ``Gcf`` to draw all the figures at once, collating them into a single pdf document, one figure per page.